File: SMarketPlace/SMarketPlace.Web/Modules/ServerTypes/Sucursales/TblSucursalRappiRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface TblSucursalRappiRow {
    LocalSap?: string;
    NombreSucursal?: string;
}

export abstract class TblSucursalRappiRow {
    static readonly idProperty = 'LocalSap';
    static readonly nameProperty = 'LocalSap';
    static readonly localTextPrefix = 'Sucursales.TblSucursalRappi';
    static readonly lookupKey = 'Sucursales.TblSucursalRappi';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TblSucursalRappiRow>('Sucursales.TblSucursalRappi') }
    static async getLookupAsync() { return getLookupAsync<TblSucursalRappiRow>('Sucursales.TblSucursalRappi') }

    static readonly deletePermission = 'Sucursales:Rappi';
    static readonly insertPermission = 'Sucursales:Rappi';
    static readonly readPermission = 'Sucursales:Rappi';
    static readonly updatePermission = 'Sucursales:Rappi';

    static readonly Fields = fieldsProxy<TblSucursalRappiRow>();
}

File: SMarketPlace/SMarketPlace.Web/Modules/Sucursales/TblSucursalRappi/TblSucursalRappiRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace SMarketPlace.Sucursales;

[LookupScript(Expiration = -1)]
[ConnectionKey("Default"), Module("Sucursales"), TableName("TBLSucursalRappi")]
[DisplayName("Tbl Sucursal Rappi"), InstanceName("Tbl Sucursal Rappi")]
[ReadPermission(PermissionKeys.Rappi)]
[ModifyPermission(PermissionKeys.Rappi)]
public sealed class TblSucursalRappiRow : Row<TblSucursalRappiRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Local Sap"), Column("LocalSAP"), Size(4), IdProperty, QuickSearch, NameProperty, ForeignKey("[dbo].[tblSucursal]", "LocalSap"), LeftJoin("jSucursales"), TextualField("NombreSuc")]
    [LookupEditor(typeof(TblSucursalRow))]
    public string LocalSap
    {
        get => fields.LocalSap[this];
        set => fields.LocalSap[this] = value;
    }
    [DisplayName("Nombre de Sucursal"), Expression("jSucursales.[NombreSuc]"), QuickSearch]
    [LookupEditor(typeof(TblSucursalRow))]
    public string NombreSucursal
    {
        get => fields.NombreSucursal[this];
        set => fields.NombreSucursal[this] = value;
    }
    public class RowFields : RowFieldsBase
    {
        public StringField LocalSap;
        public StringField NombreSucursal;
    }
    public bool TrackWithChecks { get; internal set; }
}